daemon/c_net.c

```diff
@@ -781,23 +785,25 @@ c_net_start_pre_clone_interface(c_net_interface_t *ni)
 	if (c_net_create_veth_pair(ni->veth_cont_name, ni->veth_cmld_name))
 		goto err;
 
-	DEBUG("Set root ns ipv4 address");
+	if (ni->configure) {
+		DEBUG("Set root ns ipv4 address");
 
-	/* Set IPv4 address */
-	if (c_net_set_ipv4(ni->veth_cmld_name, &ni->ipv4_cmld_addr, &ni->ipv4_bc_addr))
-		goto err;
-
-	// set subnet string
-	uint32_t ip = ntohl(ni->ipv4_cmld_addr.s_addr);
-	uint32_t mask = ~(((uint32_t)-1) >> IPV4_PREFIX);
-	struct in_addr net = { .s_addr = htonl(ip & mask) };
-	ni->subnet = mem_printf("%s/%d", inet_ntoa(net), IPV4_PREFIX);
-	IF_NULL_GOTO_ERROR(ni->subnet, err);
-
-	/* Write dhcp config for dnsmasq skip first veth (which is used in a0) */
-	if (ni->cont_offset > 0) {
-		if (c_net_write_dhcp_config(ni))
+		/* Set IPv4 address */
+		if (c_net_set_ipv4(ni->veth_cmld_name, &ni->ipv4_cmld_addr, &ni->ipv4_bc_addr))
 			goto err;
+
+		// set subnet string
+		uint32_t ip = ntohl(ni->ipv4_cmld_addr.s_addr);
+		uint32_t mask = ~(((uint32_t)-1) >> IPV4_PREFIX);
+		struct in_addr net = { .s_addr = htonl(ip & mask) };
+		ni->subnet = mem_printf("%s/%d", inet_ntoa(net), IPV4_PREFIX);
+		IF_NULL_GOTO_ERROR(ni->subnet, err);
+
+		/* Write dhcp config for dnsmasq skip first veth (which is used in a0) */
+		if (ni->cont_offset > 0) {
+			if (c_net_write_dhcp_config(ni))
+				goto err;
```

Comment:
just a quick check: do we need to free anything in err case? we allocated the "ni" struct.
By the way, I don't see an "err" marker

Reply:
err mark is there. This function does not allocate the ni struct itself, thus nothing to free here.